libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -227,6 +227,20 @@ impl NotificationHandler {
     }
 }
 
+impl Iterator for NotificationHandler {
+    type Item = CmNotification;
```

Comment:
This should probably become ConnectionManagerNotification or just Notification as part of the public API (Cm being for private implementation).